### ISSUE TITLE
How to use AstroPy's BLS with Lightkurve?

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#398]
AstroPy v3.1 has a [great implementation of BLS](http://docs.astropy.org/en/stable/api/astropy.stats.BoxLeastSquares.html).  The question begs: should we make it easy for Lightkurve users to call/use this BLS?

## Proposal

I propose to introduce two convenience methods:
* `lc.to_bls()` shall return an `astropy.BoxLeastSquares` object initialized with the LightCurve data.
* `lc.to_periodogram(method='bls', **kwargs)` shall wrap `BoxLeastSquares.power(**kwargs)` to populate and return a `lightkurve.Periodogram` object.

## Example use

Using `lc.to_bls()`:
```python
import lightkurve as lk
lc = lk.search_targetpixelfile("Kepler-10", quarter=6).download().to_lightcurve().flatten()
result = lc.to_bls().autopower(duration=0.1)
kepler10b_period = result.period[np.argmax(result.power)]
```

Using `lc.to_periodogram()`:
```python
kepler10b_period = lc.to_periodogram(method="bls").period_at_max_power
```

Thoughts, comments, ideas?

**Comment by barentsen** [comment]
Feedback from chedges and nksaunders:
* Add subclasses `LombScarglePeriodogram` and `BoxLeastSquaresPeriodogram` to preserve method-specific data/results/parameters.
* `lc.to_periodogram("bls")` must take min_period, max_period, etc...

**Comment by barentsen** [comment]
These ideas have been implemented as part of #403.  Thanks @christinahedges!! :tada:
